### PR TITLE
Style fixes.

### PR DESCRIPTION
**Pull request by anntzer**
... mostly found by pylint, plus some other stuff.
It did catch two tests in test_colorbar with the same name, one
shadowing the other :)

----

edit: also needed to update the tests that checked the exact error messages, so parametrized them at the same time.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by dopplershift** [comment]
All the more reason to work to make sure we actually run 100% of our test code. 99.09% != 100% (though I guess it will go up with this PR.)

File: lib/matplotlib/_layoutbox.py [hunk 1]
```diff
@@ -623,29 +623,21 @@ def match_margins(boxes, levels=1):
...
         print_children(child)
 
 
 def nonetree(lb):
-    '''
-    Make all elements in this tree none...  This signals not to do any more
-    layout.
-    '''
+    """
+    Make all elements in this tree none, signalling not to do any more layout.
```
[the remark this answers is not in the document]
Comment (WeatherGod) [reply lib/matplotlib/_layoutbox.py]: Also, this `none` here should probably be rendered as a literal None?

[the remark this answers is not in the document]
Comment (anntzer) [reply lib/matplotlib/_layoutbox.py]: MW says both are fine: https://www.merriam-webster.com/dictionary/signalling?utm_campaign=sd&utm_medium=serp&utm_source=jsonld
Edited none -> None.

File: lib/matplotlib/contour.py [hunk 2]
```diff
@@ -1513,39 +1505,34 @@ def _check_xyz(self, args, kwargs):
...
             x, y = np.meshgrid(x, y)
-
         elif x.ndim == 2:
-
             if x.shape != z.shape:
-                raise TypeError("Shape of x does not match that of z: found "
-                                "{0} instead of {1}.".format(x.shape, z.shape))
-
+                raise TypeError(
+                    f"Shapes of x {x.shape} and z {z.shape} do not match")
```
Comment (timhoffm) [inline lib/matplotlib/contour.py]: "Shapes of x (2, 3) and z (2, 4) do not match" would look a bit odd.

I would go for `Shapes of x and z do match: {x.shape} != {z.shape}`.

Similarly in some other places.

Comment (anntzer) [reply lib/matplotlib/contour.py]: Actually I think my formulation reads better than yours, but I'm happy to hear suggestions from others.

File: lib/matplotlib/_layoutbox.py [hunk 3]
```diff
@@ -623,29 +623,21 @@ def match_margins(boxes, levels=1):
 
 
 def seq_id():
-    '''
-    Generate a short sequential id for layoutbox objects...
-    '''
-
-    global _layoutboxobjnum
```
Comment (dstansby) [inline lib/matplotlib/_layoutbox.py]: Why has this lost the `global`?

Comment (dstansby) [inline lib/matplotlib/_layoutbox.py]: Why has this lost the `global`?

File: lib/matplotlib/contour.py [hunk 4]
```diff
@@ -1513,39 +1505,34 @@ def _check_xyz(self, args, kwargs):
...
 
         if z.ndim != 2:
-            raise TypeError("Input z must be a 2D array.")
-        elif z.shape[0] < 2 or z.shape[1] < 2:
-            raise TypeError("Input z must be at least a 2x2 array.")
-        else:
-            Ny, Nx = z.shape
+            raise TypeError(f"Input z must be 2D, not {z.ndim}D")
+        if z.shape[0] < 2 or z.shape[1] < 2:
+            raise TypeError(f"Input z must be at least a (2, 2) array, but "
```
Comment (dstansby) [inline lib/matplotlib/contour.py]:
```suggestion
            raise TypeError(f"Input z must be at least a (2, 2) shaped array, but "
```

File: lib/matplotlib/contour.py [hunk 5]
```diff
@@ -1513,39 +1505,34 @@ def _check_xyz(self, args, kwargs):
...
+                            f"has shape {z.shape}")
+        Ny, Nx = z.shape
 
         if x.ndim != y.ndim:
-            raise TypeError("Number of dimensions of x and y should match.")
-
+            raise TypeError(f"Dimensionalities of x ({x.ndim}) and y "
```
Comment (dstansby) [inline lib/matplotlib/contour.py]: I think I prefer "Number of dimensions" here (I didn't know what Dimensionalities meant until I looked at the before diff)

File: lib/matplotlib/contour.py [hunk 6]
```diff
@@ -1513,39 +1505,34 @@ def _check_xyz(self, args, kwargs):
...
+                raise TypeError(
+                    f"Shapes of x {x.shape} and z {z.shape} do not match")
             if y.shape != z.shape:
-                raise TypeError("Shape of y does not match that of z: found "
-                                "{0} instead of {1}.".format(y.shape, z.shape))
+                raise TypeError(
+                    f"Shapes of y {y.shape} and z {z.shape} do not match")
         else:
-            raise TypeError("Inputs x and y must be 1D or 2D.")
+            raise TypeError(f"Inputs x and y must be 1D or 2D, not {x.ndim}D")
```
Comment (dstansby) [inline lib/matplotlib/contour.py]: add `and {y.ndim}D` ?

File: lib/matplotlib/_layoutbox.py (hunk 3 above)
Comment (anntzer) [reply lib/matplotlib/_layoutbox.py]: you don't need `global` to read from a global (... just like you can call a module-level function from another function without declaring it as global).

File: lib/matplotlib/contour.py (hunk 6 above)
Comment (anntzer) [reply lib/matplotlib/contour.py]: we already checked above that they have the same number of dims

File: lib/matplotlib/contour.py (hunk 4 above)
Comment (anntzer) [reply lib/matplotlib/contour.py]: sure, edited

File: lib/matplotlib/contour.py (hunk 5 above)
Comment (anntzer) [reply lib/matplotlib/contour.py]: edited

File: lib/matplotlib/contour.py (hunk 2 above)
Comment (tacaswell) [reply lib/matplotlib/contour.py]: I think either is OK and tie goes to the incumbent.

**Pull request merged**